contracts/validators/request/CredentialAtomicQueryValidatorBase.sol

```diff
@@ -3,26 +3,26 @@ pragma solidity 0.8.27;
 
 import {Ownable2StepUpgradeable} from "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
 import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
-import {GenesisUtils} from "../lib/GenesisUtils.sol";
-import {ICircuitValidator} from "../interfaces/ICircuitValidator.sol";
-import {IVerifier} from "../interfaces/IVerifier.sol";
-import {IState} from "../interfaces/IState.sol";
-import {PrimitiveTypeUtils} from "../lib/PrimitiveTypeUtils.sol";
+import {GenesisUtils} from "../../lib/GenesisUtils.sol";
+import {IRequestValidator} from "../../interfaces/IRequestValidator.sol";
+import {IGroth16Verifier} from "../../interfaces/IGroth16Verifier.sol";
+import {IState} from "../../interfaces/IState.sol";
+import {PrimitiveTypeUtils} from "../../lib/PrimitiveTypeUtils.sol";
 
 /**
  * @dev Base contract for credential atomic query validators circuits.
  */
 abstract contract CredentialAtomicQueryValidatorBase is
     Ownable2StepUpgradeable,
-    ICircuitValidator,
+    IRequestValidator,
     ERC165
 {
     /// @dev Main storage structure for the contract
     /// @custom:storage-location iden3.storage.CredentialAtomicQueryValidator
     struct CredentialAtomicQueryValidatorBaseStorage {
-        mapping(string => IVerifier) _circuitIdToVerifier;
+        mapping(string => IGroth16Verifier) _circuitIdToVerifier;
         string[] _supportedCircuitIds;
-        IState state;
+        uint256[1] __gap;
```

Comment:
Need a small comment why we put this __gap

Reply:
Yes, that is what was true before we started transferring cross-chain timestamps of state and GIST.
But, as long as now Verifier should insert cross-chain messages into State contracts (Validator is not efficient to do the same instead), it would be a duplication of the link to the State contract in both Validator and Verifier. At least in terms of contract init, It's better only one component to keep the link.

Reply:
To me it looks strange and a place for a calling contract to make a mistake. The less possibilities to make a mistake - the better.

Reply:
So please revert this part and remove passing of state contract address from the verify function.

Reply:
Done